src/app.js

```diff
@@ -184,56 +177,57 @@ var run = function () {
   var editor = new Editor(document.getElementById('input'))
 
   // ---------------- FilePanel --------------------
+  // TODO: All FilePanel related CSS should move into file-panel.js
+  // app.js provides file-panel.js with a css selector or a DOM element
+  // and file-panel.js adds its elements (including css), see "Editor" above
   var css = csjs`
     .filepanel    {
       display     : flex;
       width       : 200px;
     }
   `
-  var filepanel = document.querySelector('#filepanel')
-  filepanel.className = css.filepanel
+  var filepanelContainer = document.querySelector('#filepanel')
+  filepanelContainer.className = css.filepanel
   var FilePanelAPI = {
     createName: createNonClashingName,
     switchToFile: switchToFile,
     event: this.event
   }
-  var el = new FilePanel(FilePanelAPI, files)
-  filepanel.appendChild(el)
-  var api = el.api
+  var filePanel = new FilePanel(FilePanelAPI, files)
+  // TODO this should happen inside file-panel.js
+  filepanelContainer.appendChild(filePanel)
 
-  api.register('ui', function changeLayout (data) {
+  filePanel.events.register('ui-hidden', function changeLayout (isHidden) {
```

Comment:
So instead of having a `ui` event and then having the event or "payload" with the structure like for example:
* `{ type: 'minimize', value: true,  ... }`
* `{ type: 'maximize', value: true,  ... }`
* `{ type: 'show', value: true,  ... }`
* `{ type: 'hide', value: true,  ... }`
* ...

instead there would be a seperate event for each of those, so event `ui` and `{ type: 'hide', value: true,  ... }`
becomes event `ui-hidden` and `true`, right?

If that's the case, then ok, i'm fine with the change. It's probably a matter of personal taste. I usually like to send messages around that have a "type" field.